src/Core/Managed/Shared/Extensibility/Implementation/SimpleMetricStatisticsAggregator.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Represents aggregator for a single time series of a given metric.
+    /// </summary>
+    internal class SimpleMetricStatisticsAggregator
```

Comment:
I need to think about Metric/MetricAggregator naming...

You're almost right. There are two aggregators, so to speak. MetricAggregator is really a "handle" or a proxy to te actual aggregator. You can hold on to it aslong as you like and operating on it will be proxied to actual aggregator which is SimpleMetricStatisticsAggregator class.
Note that metric data may be aggregated in a number of ways. Extracting/computing simple stats is just a way. You can also do t-digest and some other fun stuff too. Lets chat more about it.